### PR TITLE
Introduce user roles

### PR DESCRIPTION
**Pull request by dionyziz**

### CODE REVIEW

**Comment by dionyziz** [comment]
This is needed to add profiling abilities to developers. It will also help guard permissions for executing migrations and running unit tests.

File: database/migration/20140516235600_add_user_default_role.php [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?php
```
Comment (pkakelas) [inline database/migration/20140516235600_add_user_default_role.php:1]: Niice, I've never thought that we can use the migrate function independently

File: models/user.php [hunk 2]
```diff
@@ -152,6 +155,9 @@ protected function onBeforeSave() {
             if ( !isset( $this->boturl ) ) {
                 $this->boturl = '';
             }
+            if ( $this->role < 0 || !isset( $this->role ) ) {
```
Comment (pkakelas) [inline models/user.php:158]: How can this happen?

Comment (dionyziz) [inline models/user.php:158]: Negative values cannot happen unless they are set specifically; this is just a safeguard. Unset roles can occur if a new user is constructed and no role is assigned.

Comment (pkakelas) [inline models/user.php:158]: why don't you make $this-role a boolean?

Comment (dionyziz) [inline models/user.php:158]: For future extensibility; we may want to add other roles inbetween user and developer, e.g. "admin" or "contributor" that don't have developer roles, but have some increased user abilities.

**Pull request merged**